test/anacondalib.py

```diff
@@ -60,6 +76,31 @@ def setUp(self):
             self.allow_browser_errors(".*client closed.*")
             self.allow_browser_errors(".*Server has closed the connection.*")
 
+    def _execute(self, cmd):
+        return subprocess.check_call(cmd, stderr=subprocess.STDOUT, shell=True)
+
+    def add_disk(self, size, backing_file=None):
+        image = self._create_disk_image(size, backing_file=backing_file)
+        self._execute(f"virt-xml -c qemu:///session {self.machine.label} --update --add-device --disk {image},format=qcow2")
+
+        if self.is_nondestructive():
+            self.addCleanup(self.rem_disk, image)
+
+        return image
+
+    def rem_disk(self, disk):
+        self._execute(f"virt-xml -c qemu:///session {self.machine.label} --update --remove-device --disk {disk}")
+        os.remove(disk)
+
+    def _create_disk_image(self, size, image_path=None, quiet=False, backing_file=None):
+        if not image_path:
+            _, image_path = tempfile.mkstemp(suffix='.qcow2', prefix=f"disk-anaconda-{self.machine.label}")
+        quiet = "-q" if quiet else ""
+        backing_file = f"-o backing_file={backing_file},backing_fmt=qcow2" if backing_file else ""
+        size = f"{size}G" if size else ""
+        self._execute(f"qemu-img create -f qcow2 {backing_file} {quiet} {image_path} {size}")
```

Comment:
Ah, you are playing a little dirty shell trick here with an empty size ... :grin: